code/game/gamemodes/wizard/artefact.dm

```diff
@@ -12,8 +12,18 @@
 	force = 15
 	throwforce = 10
 	w_class = 3
-	var/charged = 1
 	hitsound = 'sound/weapons/bladeslice.ogg'
+	var/charged = 1
+	var/spawn_type = /obj/singularity/narsie/wizard
+	var/spawn_amt = 1
+
+/obj/item/weapon/veilrender/attack_self(mob/user as mob)
+	if(charged)
+		new /obj/effect/rend(get_turf(usr), spawn_type, spawn_amt)
+		charged = 0
+		user.visible_message("<span class='userdanger'>[src] hums with power as [usr] deals a blow to reality itself!</span>")

```

Comment:
The "to reality itself" part should change based on a var, as the veal render used "to hunger itself" there.